tests/ballerina-compiler-api-test/src/test/resources/test-src/expressions_test.bal

```diff
@@ -358,3 +358,6 @@ function fooFn() returns future<int> {
 function barFn() returns int {
     return 10;
 }
+
+const int A = 10;
+const int B = 20 + 30;
```

Comment:
@sanjana I have a doubt on the original issue (https://github.com/ballerina-platform/ballerina-lang/issues/36831) though.
Say I want to extract the constant. The extracted constant can be either the exact singleton value `30` or it can have the wider basic type node `int`. As of the current implementation, it only allows `int`. But @SandaruJayawardana is doing a complete revamp for constant expressions and both have to be allowed. What should be the user experience for this one?

```bal
const 30 A = 10 + 20; // valid
const int A = 10 + 20; // also valid
```

Reply:
The issue is with the expression, right?

I think the behaviour should be consistent for both RHS expressions in 

```ballerina
const int A = 10;
const int B = 20 + 30;
```

If the type of `10` is `int`, it should be the same for `20 + 30` also. But if the type for `20 + 30` is singleton `50`, the type for `10` should also be singleton `10`. Feels like the singleton approach is more accurate, but since this applies to all literals, may result in a lot of new types being created.

For example, even with `int i = 10;`, the type of `10` will then be the singleton. 

I think we need to discuss this and come to a conclusion and then fix across the compiler for all literals.

Reply:
Will be holding this PR until the discussion initiated at #38250 is finalized.